HARK/ConsumptionSaving/ConsAggShockModel.py

```diff
@@ -550,11 +550,325 @@ def getMrkvNow(self):  # This function exists to be overwritten in StickyE model
+    in each macroeconomic state (bad=0, good=1), rather than aggregate capital as
+    a function of previous aggregate capital.  This choice was made so that some
+    of the code from HARK's other HA-macro models can be used.
+    '''
+    
+    def __init__(self, **kwds):
+        '''
+        Make a new instance of the Krusell-Smith type.
+        '''
+        params = init_KS_agents.copy()
+        params.update(kwds)
+
+        AgentType.__init__(self, pseudo_terminal=False, **params)
+
+        # Add consumer-type specific objects
+        self.time_vary = []
+        self.time_inv = ['DiscFac', 'CRRA',]
+        self.poststate_vars = ['aNow','EmpNow']
+        self.solveOnePeriod = solveKrusellSmith
+        self.update()
+        
+    def preSolve(self):
+        self.update()
+        self.preComputeArrays()
+        
+    def update(self):
+        '''
+        Construct objects used during solution from primitive parameters.
+        '''
+        self.makeGrid()
+        self.updateSolutionTerminal()
+        
+    def getEconomyData(self, Economy):
+        '''
+        Imports economy-determined objects into self from a Market.
+
+        Parameters
+        ----------
+        Economy : KrusellSmithEconomy
+            The "macroeconomy" in which this instance "lives".
+
+        Returns
+        -------
+        None
+        '''
+        self.T_sim = Economy.act_T                   # Need to be able to track as many periods as economy runs
+        self.kInit = Economy.KSS                     # Initialize simulation assets to steady state
+        self.MrkvInit = Economy.MrkvNow_init         # Starting Markov state for the macroeconomy
+        self.Mgrid = Economy.MSS*self.MgridBase      # Aggregate market resources grid adjusted around SS capital ratio
+        self.AFunc = Economy.AFunc                   # Next period's aggregate savings function
+        self.DeprFac = Economy.DeprFac               # Rate of capital depreciation
+        self.CapShare = Economy.CapShare             # Capital's share of production
+        self.LbrInd = Economy.LbrInd                 # Idiosyncratic labor supply (when employed)
+        self.UrateB = Economy.UrateB                 # Unemployment rate in bad state
+        self.UrateG = Economy.UrateG                 # Unemployment rate in good state
+        self.ProdB = Economy.ProdB                   # Total factor productivity in bad state
+        self.ProdG = Economy.ProdG                   # Total factor productivity in good state
+        self.MrkvIndArray = Economy.MrkvIndArray     # Transition probabilities among discrete states
+        self.MrkvAggArray = Economy.MrkvArray        # Transition probabilities among aggregate discrete states
+        self.addToTimeInv('Mgrid', 'AFunc', 'DeprFac', 'CapShare', 'UrateB', 'LbrInd',\
+                          'UrateG', 'ProdB', 'ProdG', 'MrkvIndArray', 'MrkvAggArray')
```

Comment:
See #764 . If those aspects of the economy that needed to be brought into the Agent were available in a list (as most of them are here), then it would be possible to write this method generically as a loop over those values, rather than writing out each one individually.

This is common across all uses of getEconomyData